lib/BaseClient.php

```diff
@@ -67,6 +67,8 @@ protected function internalStartSearch(Fingerprint $ft): SearchFuture
         Lib::get()->Pex_StartSearchRequest_SetFingerprint($startReq, $buffer, $status);
         Error::checkStatus($status);
 
+        Lib::get()->Pex_StartSearchRequest_SetType($startReq, $type);
+
```

Comment:
I presume there's no `Error::checkStatus` here because the `SetType` can't possibly fail? Is that likely to be true forever?

Reply:
Yeah, there's nothing really we can do with the search type. It's different for fingerprint where we can actually unserialize it and check the validity, but we can't check the type until we're actually making the request.